app/src/main/java/com/example/poste/models/Folder.java

```diff
@@ -60,6 +69,34 @@ public String getUserPermission() {
         return userPermission;
     }
 
+    /**
+     * Adds a new post the given folder
+     *
+     * @param folder folder to add the new post to
+     */
+    public Post createNewPost(Folder folder) {
```

Comment:
Similar to what was discussed above:

This method should probably only be called once we have all the appropriate fields: Title, Description, URL, and the request should contain all the fields necessary. You make the API call in 82 with newPost.getDescription(), newPost.getUrl(), but that has just been built without these fields -- they'll never be set. We need to retrieve these from a client-side page (activity)

Take a look at how Login or Register work; an activity is made with fields for the user to fill in, and once they're filled in, the data is retrieved and validated, then sent to the API

This is a great start, however. I would also encourage you to handle failures in some way; otherwise, if the API call fails, nothing will happen, and the user won't know why.

Reply:
Rewritten as directed in discussion this morning